internal/worker/logger/logger.go

```diff
@@ -93,6 +95,17 @@ func (l *loggerWorker) setLogging() {
 		logger.Debugf("reconfiguring logging from %q to %q", l.lastConfig, loggingConfig)
 		context := l.config.Context
 		context.ResetLoggerLevels()
+
+		// check if loggingConfig string includes "juju.security", which is not allowed to change
+		// as it would allow changing the security log level to a less restrictive level
+		// and potentially miss important security events
+		if strings.Contains(loggingConfig, "juju.security") {
+			logger.Warningf("logging config includes juju.security, which is not allowed")
+			// Try to reset to what we had before
+			_ = context.ConfigureLoggers(l.lastConfig)
+			return
+		}
```

Comment:
We could actually just prevent this during setting the model-config.

I'd drop this piece of code from this PR, and propose a follow up with those changes. That way we can ensure that the spec work is correct and not be concerned about if the loggers can be overridden.